src/main/scala/tilelink/Monitor.scala

```diff
@@ -31,18 +33,26 @@ object TLMonitor {
   }
 }
 
+@chiselName
 class TLMonitor(args: TLMonitorArgs, monitorDir: MonitorDirection = MonitorDirection.Monitor) extends TLMonitorBase(args)
 {
+  require (args.edge.params(TLMonitorStrictMode) || (! args.edge.params(TestplanTestType).formal))
+
   val cover_prop_class = PropertyClass.Default
   val desc_text = "Placeholder"
 
-  /*def assert(cond: Bool, message: String): Unit = {
+  def monAssert(cond: Bool, message: String): Unit = if (monitorDir == MonitorDirection.Monitor)
```

Comment:
Need to move monAssert and assume to FormalUtils...

Reply:
This is a nit but stylistically I'd write this as:
```scala
def monAssert(cond: Bool, message: String): Unit =
  if (monitorDir == MonitorDirection.Monitor) {
    assert(cond, message)
  } else {
    Property(monitorDir, cond, message, PropertyClass.Default, desc_text)
  }
```

In particular, Scala-style dictates that curly-braces go on the same line as the declaration they represent (https://docs.scala-lang.org/style/nested-blocks.html#curly-braces). It's usually harmless to put it on the next line, but apparently it can cause issues with the parser.

I would also add a comment explaining why you have `monAssert` instead of just `assert` (making it "flippable") just for future people looking at this code.

Reply:
I also suggested that this definition should go somewhere more common (maybe formal.utils) instead of making it special to Monitor.  I also would think there should be a better way to be more future compatible, but haven't think too hard about it yet.  Not sure if @jackkoenig  have better suggestions.

Reply:
Jacob's suggested change is gonna go into a future PR

Reply:
I implemented Jack's suggested changes.